apps/editing-toolkit/editing-toolkit-plugin/wpcom-block-description-links/src/index.ts

```diff
@@ -8,21 +8,11 @@ const addBlockSupportLinks = (
 	},
 	name: string
 ) => {
-	/**
-	 * Adjust the block name to apply link to InnerBlocks. This gets reset at the end
-	 */
-	const applyToChildren = [
-		'core/columns',
-		'core/social-links',
-		'core/buttons',
-		'jetpack/contact-form',
-	].includes( settings[ 'parent' ]?.toString() );
-
-	if ( applyToChildren ) {
-		name = settings[ 'parent' ]?.toString();
-	}
+	// If block has a parent, use the parents name in the switch. This will apply the link to all nested blocks.
+	const isChild = settings[ 'parent' ];
+	const blockName = isChild ? settings[ 'parent' ].toString() : name;
```

Comment:
Thanks for removing the mutation out of `name`. 

I did notice that we're also mutating the `settings` argument, though (inside the `case` blocks). Is that another mutation you'd we'd like to avoid?

Reply:
No, that is the intended design to overwrite the `settings.description` to include the inline support link. There is no other hook or filter in core that gives us this ability so this extension mutates the description and gives it back to the editor.